### PR TITLE
Upgrade to Babel 7

### LINKED ISSUE #65: Upgrade to Babel 7
Babel 7 is out so we should catch up. 😄 

https://babeljs.io/blog/2018/08/27/7.0.0

### PR DESCRIPTION
**Pull request by pronebird**
This PR upgrades Babel from v6 to v7 and bumps Electron to v3.0 along with a few other deps

Fixes #65

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,16 +1,13 @@
...
+        "electron": "3.0"
+      }
+    }], "@babel/preset-react"
   ],
   "plugins": [
-    "transform-decorators-legacy",
-    "transform-class-properties",
-    "transform-object-rest-spread",
-    "transform-runtime"
+    ["@babel/plugin-proposal-decorators", { "legacy": true }],
```
Comment (pronebird) [inline .babelrc]: When using `legacy` decorators, `loose` is important to use on `class-properties` extension according to official docs: https://babeljs.io/docs/en/babel-plugin-proposal-decorators

File: package.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
   "repository": "https://github.com/jschr/electron-react-redux-boilerplate",
   "license": "MIT",
   "dependencies": {
-    "babel-runtime": "^6.26.0",
```
Comment (pronebird) [inline package.json]: I didn't find any need for babel-runtime in modern Node.js. I may be wrong, but so far so good :)

**Comment by jschr** [comment]
Amazing! Thanks for doing this. 

I'll check this out this out locally later today!

**Comment by pronebird** [comment]
@jschr 👍 Just pushed the final change to migrate from `.babelrc` to new `babel.config.js`.

**Comment by pronebird** [comment]
@jschr let me check. it must be `@babel/runtime` these days. My installation didn't require that.

**Comment by pronebird** [comment]
@jschr can't reproduce that.

Please run the following and try again:

```
rm package-lock.json
rm -rf build
rm -rf node_modules
npm install
npm run develop
```

**Comment by jschr** [comment]
Works! I didn't clear the `build` dir before so that must have been why.

Thanks!

**Pull request merged**